Update linting for Angular 17

RadLikeWhoa opened the pull request.

RadLikeWhoa committed:
feat(angular): udpate linting config #GUILD-339

RadLikeWhoa committed:
feat(angular): update dependencies #GUILD-339

kelvinlouis commented on packages/eslint-config-angular/index.js:
```diff
@@ -40,6 +40,7 @@ module.exports = {
     '@angular-eslint/prefer-output-readonly': 'error',
     '@angular-eslint/relative-url-prefix': 'error',
     '@angular-eslint/use-injectable-provided-in': 'error',
+    '@angular-eslint/prefer-standalone-component': 'warn',
```
What's the reason from going from error to warn? Or was there no error at all in the first place?

RadLikeWhoa replied:
Default config does not include this rule. Only a warning since there are valid cases for non-standalone components.

kelvinlouis commented on packages/eslint-config-angular/index.js:
```diff
@@ -50,4 +51,41 @@ module.exports = {
+        '*.ts',
+      ],
+      extends: [
+        'plugin:@angular-eslint/template/process-inline-templates'
```
dangling comma

kelvinlouis commented on packages/eslint-config-angular/index.js:
```diff
@@ -50,4 +51,41 @@ module.exports = {
+        '*.html',
+      ],
+      extends: [
+        'plugin:@angular-eslint/template/recommended'
```
dangling comma

kelvinlouis commented on packages/eslint-config-angular/index.js:
```diff
@@ -50,4 +51,41 @@ module.exports = {
+              'ATTRIBUTE_BINDING',
+              'INPUT_BINDING',
+              'TWO_WAY_BINDING',
+              'OUTPUT_BINDING'
```
dangling comma

kelvinlouis reviewed (commented):
Did you check with #yoo_devs_angular about introducing the new rules?

RadLikeWhoa committed:
fix: fix formatting #GUILD-339

RadLikeWhoa committed:
feat: update versions #GUILD-339

RadLikeWhoa committed:
Publish

 - @yoo-digital/eslint-config-angular@17.0.0-canary.1
 - @yoo-digital/eslint-config-base@1.2.4-canary.0
 - @yoo-digital/eslint-config-react@1.2.4-canary.0

kelvinlouis commented:
We should also try to remove the base setup from the angular package.
This means removing it from the extends statement and also from the dependencies in the package.json.
After doing these changes `npm i` is going to be necessary again.

From there on the projects themselves will have to extend and install both packages explicitly. 

@RadLikeWhoa

RadLikeWhoa committed:
chore: remove base dependency from angular package #GUILD-339
